crates/uv-trampoline/src/bounce.rs

```diff
@@ -240,10 +250,18 @@ fn read_trampoline_metadata(executable_name: &Path) -> (TrampolineKind, PathBuf)
         parent_dir.join(path)
     };
 
-    // NOTICE: dunce adds 5kb~
-    let path = dunce::canonicalize(path.as_path()).unwrap_or_else(|_| {
-        error_and_exit("Failed to canonicalize script path");
-    });
+    let path = if !path.is_absolute() || matches!(kind, TrampolineKind::Script) {
+        // NOTICE: dunce adds 5kb~
+        // TODO(john): In order to avoid resolving junctions and symlinks for relative paths and
+        // scripts, we can consider reverting https://github.com/astral-sh/uv/pull/5750/files#diff-969979506be03e89476feade2edebb4689a9c261f325988d3c7efc5e51de26d1L273-L277.
```

Comment:
I don't quite follow this TODO

Reply:
In this code, if this is a trampoline `Script` or a relative path, we call `dunce::canonicalize`. On `main` this was true, but we also called it for absolute paths that were `Python` trampolines (because we always called it). So this TODO is a path to avoid resolving symlinks and junctions in all cases.